src/reanimated2/interpolation.ts

```diff
@@ -1,149 +1,165 @@
+      return rightEdgeOutput;
+    case Extrapolatation.EXTEND:
     default:
       return val;
   }
 }
 
-function isExtrapolate(value) {
+function isExtrapolate(value: string): boolean {
   'worklet';
 
   return (
-    value === Extrapolate.EXTEND ||
-    value === Extrapolate.CLAMP ||
-    value === Extrapolate.IDENTITY
+    value === Extrapolatation.EXTEND ||
+    value === Extrapolatation.CLAMP ||
+    value === Extrapolatation.IDENTITY
   );
 }
 
-function validateType(type) {
+// validates extrapolations type
+// if type is correct, converts it to ExtrapolationConfig
+function validateType(type: ExtrapolationType): Required<ExtrapolationConfig> {
   'worklet';
-
-  const EXTRAPOLATE_ERROR_MSG = `Reanimated: config object is not valid, please provide valid config, for example:
-    interpolate(value, [inputRange], [outputRange], {
-    extrapolateLeft: 'clamp',
-    extrapolateRight: 'extend',
-  })`;
-
-  const EXTRAPOLATE_ERROR = (
-    extrapolate
-  ) => `Reanimated: not supported value for "${extrapolate}" \nSupported values: ["extend", "clamp", "identity"]\n Valid example:
-    interpolate(value, [inputRange], [outputRange], {
-      ${extrapolate}: 'clamp',
-  })`;
-
-  type = type ?? 'extend';
-
-  // eslint-disable-next-line no-prototype-builtins
-  const hasExtrapolateLeft = type.hasOwnProperty('extrapolateLeft');
-  // eslint-disable-next-line no-prototype-builtins
-  const hasExtrapolateRight = type.hasOwnProperty('extrapolateRight');
-
-  if (
-    typeof type === 'object' &&
-    ((Object.keys(type).length === 2 &&
-      !(hasExtrapolateLeft && hasExtrapolateRight)) ||
-      (Object.keys(type).length === 1 &&
-        !(hasExtrapolateLeft || hasExtrapolateRight)) ||
-      Object.keys(type).length > 2)
-  ) {
-    throw new Error(EXTRAPOLATE_ERROR_MSG);
-  }
-
-  if (typeof type === 'object') {
-    if (hasExtrapolateLeft && !isExtrapolate(type.extrapolateLeft)) {
-      throw new Error(EXTRAPOLATE_ERROR('extrapolateLeft'));
-    }
-
-    if (hasExtrapolateRight && !isExtrapolate(type.extrapolateRight)) {
-      throw new Error(EXTRAPOLATE_ERROR('extrapolateRight'));
+  // initialize extrapolationConfig with default extrapolation
+  const extrapolationConfig: Required<ExtrapolationConfig> = {
+    extrapolateLeft: Extrapolatation.EXTEND,
+    extrapolateRight: Extrapolatation.EXTEND,
+  };
+
+  if (typeof type === 'string') {
+    if (!isExtrapolate(type)) {
+      throw new Error(
+        `Reanimated: not supported value for "interpolate" \nSupported values: ["extend", "clamp", "identity", Extrapolatation.CLAMP, Extrapolatation.EXTEND, Extrapolatation.IDENTITY]\n Valid example:
+        interpolate(value, [inputRange], [outputRange], "clamp")`
+      );
     }
+    extrapolationConfig.extrapolateLeft = type as Extrapolatation;
+    extrapolationConfig.extrapolateRight = type as Extrapolatation;
+    return extrapolationConfig;
   }
 
-  if (typeof type === 'string' && !isExtrapolate(type)) {
```

Comment:
I added type for extrapolation config, so now the user can see what values are supported, and I thought that it will be no longer necessary to prompt him what values are supported with those error messages. If you find it relevant I can revert it.

Reply:
But not everyone uses TS.

Reply:
I actually think it's a fair assumption to rely on TypeScript for type checking parameters.

1. It's a lot faster if you don't have to check for valid types (and even do function calls/equality comparisons) each time you call a function, especially if that function is a hot spot (interpolate? might get called more often than 60 times a second)
2. Almost everyone in React Native uses VSCode which _automatically_ has TypeScript support, even if you use `.js`, it will let you know that the type passed in is invalid (unless it's `any`, which I guess can happen too) - some people might use WebStorm or even Vim, but that's a minority, if you want to support everyone then I guess it's fair to leave those things in


just my take on this 🤷‍♂️